controller/src/controller/services/source/gar.py

```diff
@@ -144,11 +149,20 @@ def menu_images(self) -> list[MenuImage]:
         Returns
         -------
         list of MenuImage
-            All known images.
+            All known images meeting filter criteria.
         """
+        # This is kind of clunky.  I feel like the implementation of the
+        # storage layer threw away the tags too soon.  That said, adding a
+        # to_rsptag() method to MenuImage lets us do this in a straightforward
+        # if not very elegant way.
+        all_tags = [x.to_rsptag() for x in list(self._images.all_images())]
+        all_coll = RSPImageTagCollection(all_tags)
+        filtered_coll = all_coll.apply_policy(self._image_filter)
+        filtered_tags = list(filtered_coll.all_tags())
```

Comment:
I think this is too clunky. It's a lot cleaner semantically to add a `filter` method to `RSPImageCollection` instead. That means either a bit of code duplication or moving filtering of each class of images into a generic helper function, but it means we have a clean abstract data type and aren't doing complicated workarounds. As above, this should pass in `datetime.now(tz=UTC)` as the basis for tag ages.

Reply:
I don't understand this comment.  The problem I'm trying to solve here is that the storage layer gives us back a collection that, when iterated, emits `MenuImage`, but the thing we want for the dropdown is a collection of RSPImageTags.  So either I need to rewrite the storage layer to emit Tags, or I need a way to recover a tag from a MenuImage.  Or I need to pack the Tag directly into the MenuImage but then serialization becomes harder and we end up with duplicated fields.

Reply:
I feel like there's too much of an impedance mismatch between GAR and Docker drivers here, but I also don't feel like I really want to embark on that refactor yet.

The Docker v2 API does at least exist now, although it is not (I think) fully supported by Docker Hub(!!) and I have no idea whether Nexus et al support it.  But I think maybe we should defer this bit of work until (and unless) we tackle Docker registries again.

Reply:
Oh, never mind.  Add a `filter()` to `RSPImageCollection` too, that does the right thing, and maybe factor out the actual policy application to an external function.  I understand now.